dashboard/test/ui/features/dance/age_filter.feature

```diff
@@ -81,3 +81,47 @@ Feature: Dance Lab Age Filter
     #synthesize is available on local and shapeofyou_edsheeran is available on test
     And I see option "Synthesize" or "Ed Sheeran - Shape of You" in the dropdown "#song_selector"
 
+  Scenario: Song selector is hidden when initializing with teacher flag on and teacher flag stays on after level complete
+    Given I am on "http://studio.code.org/s/allthethings/stage/37/puzzle/1?noautoplay=true&songfilter=on"
+    And I rotate to landscape
+    And I wait for the page to fully load
+    And I wait for 3 seconds
+    And I wait until I don't see selector "#p5_loading"
+    And element ".age-dialog" is not visible
+    And I close the instructions overlay if it exists
+
+    #Local PG-13 option should not be visible after filter in any environment
+    And I do not see "synthesize" option in the dropdown "#song_selector"
+    #Test PG-13 option should not be visible after filter in any environment
+    And I do not see "shapeofyou_edsheeran" option in the dropdown "#song_selector"
+
+    #Run for two measures until level success
+    Then I click selector "#runButton" once I see it
+    And I wait until element ".congrats" is visible
+    And I press "continue-button"
+    # Make sure continue takes us to next level
+    And I wait until current URL contains "/stage/37/puzzle/2"
+    And I wait for the page to fully load
+    And I close the instructions overlay if it exists
+    #Local PG-13 option should not be visible after filter in any environment
+    And I do not see "synthesize" option in the dropdown "#song_selector"
```

Comment:
FYI, you should be able to see the restricted song list locally (and in circleci) now if you have s3 and cloudfront keys set.